### ISSUE TITLE
Sonos: Hide duplicate devices, allow shortened device name and other improvements

### ISSUE THREAD

**Issue by aglavic** [jellyfin/jellyfin-plugin-dlna#118]
Hi, thanks for the nice work keeping the DLNA plugin alive.

I'm using it mainly to stream music to Sonos devices and it mostly works well. There are a few things I don't get a handle on, yet, that might either require different configuration or modification to the plugin. Maybe someone can help me with this or point me to a way how to implement a better solution.

My issues:
1. I have multiple devices that have more then one IP address but belong to the same playback device (e.g. stereo group). Other libraries like the python [soco](https://github.com/SoCo/SoCo) can detect this and find the associated master device. I don't know what API call would be needed to detect such device and hide it in the DLNA device list. Without this I see 12 different devices instead of the actual 4 play groups I have, which makes it quite confusing (there are some 5.1 devices with 4 IPs in there).
2. The name shown for Sonos devices is extremely long and confusing as it includes some unique identifyier instead of just the type and room name. (e.g. "IP - Sonos Port - RICON_.{very long ID} - Wohnzimmer" instead of e.g. "Wohnzimmer | Port (IP)"). It would be great if the naming convention could be defined in the profile file somehow using placeholders.
3. Playback Queue: There seems to be some sort of playback queue. I'm not sure how it works, but if I play a full Album it is played correctly. But I didn't find a way to add something to that playback queue. The "play next" option seems to just replace the queue. Am I doing something wrong here?
4. It seems the playback gain option is not being used for DLNA streaming. Is there a way I can make Jellyfin overwrite the playback gain value used by Sonos with the Jellyfin Album Gain?

For issue 1 and 2 I created a personal work-around in [this fork](https://github.com/jellyfin/jellyfin-plugin-dlna/compare/master...aglavic:jellyfin-plugin-dlna:master). I'm basically checking for Sonos device in AddDevice, change the device name definition and filter for certain local IPs to be ignored as I don't know how to query for the device master/slave property. I've also added a profile for a Sonos device, not sure if that's correct as it's just based on one of the other examples.

I'm happy to try to implement a solution myself and create a pull request, if someone can get me the right pointer. I don't have any experience in coding for Jellyfin or in C#, so I might be missing something obvious.

**Comment by disgustipated** [comment]
For number 2, is there a way to modify the name from a Sonos control? My Onkyo allows renaming the device from within its app, which changes how the device name is broadcast.

I can see how number 1 would be useful with a plethora of devices that you dont need/want to see. With your fork it looks like youre hard coding specific IPs. This might be useful to have controls in the settings page to filter by a list of ignored devices. IP might not be the best as some users might use dhcp, a combination of the device id similar to how profiles are matched might work better. That might depend on how your multiple devices are broadcast.

**Comment by aglavic** [comment]
The only name I can change in the device configuration is the "Wohnzimmer" bit. If I recall correctly the plugin build the device name from various parts of the DLNA information and the long ID is a different one that is fixed by the Sonos hardware.

I agree, having some generic way of filtering that is configured manually in the back-end seems a relatively straight forward solution that covers other use cases. Maybe you can just use the unique IDs and give the user a drop-down to choose from chowing the display names for each ID.

**Comment by aglavic** [comment]
Nope, still seems to behave the same. I'm not quite sure how to test this with my limited understanding of how it works under the hood.
If anyone could point me to the deocumentation/code where the adding of songs to the queue happens, I could try to investigate further.
